posts/2020-03-31-Rust-survey-2019.md

```diff
@@ -0,0 +1,216 @@
+
+## IDEs and Tooling - a Closer Look
+
+IDE support for Rust was also cited as a barrier to adoption.
+
+When we asked users what editors they use, Vim and VSCode were the most popular by far, followed by Intellij.
+
+We also asked what IDE setups users used:
+* 43.3% indicated RLS
+* 21.7% indicated Intellij
+* 15.2% indicated Rust-analyzer
+* 12.4% indicated No (or CTAGS)
+* 4.2% indicated Only Racer
+
+As for platforms that users develop on - Linux and Windows continue to dominate. 
+* 55% of Rust users develop on Linux
+* 24% develop on Windows
+* 23% develop on macOS
+
+![Which Rust version do you use for your applications?](images/2020-03-RustSurvey/16-Which-Rust-Version.png)
+
+The vast majority of users use the current stable version of Rust (63%). Surprisingly, the number of users using the Nightly compiler in their workflow is down at 20%. Last year it was at over 56%.
+
+[ADD COPY/analysis]
+
+## Learning Curve - a Closer Look
+
+Rust is well known for its significant learning curve. 
+
+![How would you rate your Rust expertise](/images/2020-03-RustSurvey/13-Rust-Expertise-and-Time-In-Rust.png)
+
+We asked survey respondents to rate their own Rust expertise.
+
+![How long did it take to be productive](/images/2020-03-RustSurvey/8-How-Long-To-Be-Productive.png)
+
+Next, in order to better understand the learning curve, we asked how long it took users to feel that they were productive with Rust.
+
+About 37% of Rust users felt productive in Rust in less than a month of use - this is not too different from the percentage last year (40%). Over 70% felt productive in their first year. Unfortunately, like last year, there is still a struggle among users - 21% indicated they did not yet feel productive.
+
+![How long have you been unproductive](/images/2020-03-RustSurvey/9-How-Long-Unproductive.png)
+
+Additionally, many users indicated that they were not feeling comfortable in Rust - even after over 3 months of use. This is likely a result of Rust's high learning curve.
+
+This may also be an issue with language complexity, but it might also be related to tooling or other obstacles (integrated development environments in particular). This is a good place for us to investigate through follow-up surveys and other work.
+
+![Expertise level of respondents who don't feel productive yet](/images/2020-03-RustSurvey/22-unproductive-expertise.png)
+
+As a point of interest, we took the subset of users who don't feel productive yet and plotted their ratings of their Rust expertise. This indicates that people who don't feel productive had low to intermediate levels of expertise - which are the groups that need the most support from our learning materials, documentation, and more.
+
+## Interoperability - a Closer Look
+
+Over the years some users have expressed a desire for Rust to be more interoperable with other languages.
+
+![What languages would you want to use with Rust](/images/2020-03-RustSurvey/23-interoperability-languages.png)
```

Comment:
Was this an exclusive question? I seem to recall being able to select more than one -- in that case a pie chart would be the wrong choice I think? Not sure exactly what the percentages mean (i.e. they're of some sort of "total number of selected choices" not "people selecting")